akka-actor/src/main/scala/akka/actor/FSM.scala

```diff
@@ -648,13 +653,18 @@ trait FSM[S, D] extends Actor with Listeners with ActorLogging {
         this.nextState = null
       }
       currentState = nextState
-      val timeout = if (currentState.timeout.isDefined) currentState.timeout else stateTimeouts(currentState.stateName)
-      if (timeout.isDefined) {
-        val t = timeout.get
-        if (t.isFinite && t.length >= 0) {
-          import context.dispatcher
-          timeoutFuture = Some(context.system.scheduler.scheduleOnce(t, self, TimeoutMarker(generation)))
-        }
+
+      def scheduleTimeout(d: FiniteDuration): Some[Cancellable] = {
+        import context.dispatcher
+        Some(context.system.scheduler.scheduleOnce(d, self, TimeoutMarker(generation)))
+      }
+
+      currentState.timeout match {
+        case SomeMaxFiniteDuration                    ⇒ // effectively disable stateTimeout
```

Comment:
effectively what I need to express "such big timeout that the akka scheduler would blow up anyway". 
Thought using MaxValue would be cleanest as marker value. I could dig into the scheduler and find it's max and cut-off anything above it to => "effectively never", but that sounds a bit more magic than picking MaxValue as my marker.

Note: Users are not expected to use it, they'd do `forMax(Inf)` which I encode as this MaxValue marker. (Can't change types, if I could we'd change things to Duration from FiniteDuration and we'd be good).